src/Sentry/Internal/DefaultSentryStructuredLogger.cs

```diff
@@ -0,0 +1,131 @@
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Protocol.Envelopes;
+
+namespace Sentry.Internal;
+
+internal sealed class DefaultSentryStructuredLogger : SentryStructuredLogger
+{
+    private readonly IHub _hub;
+    private readonly IInternalScopeManager _scopeManager;
+    private readonly SentryOptions _options;
+    private readonly ISystemClock _clock;
+
+    private readonly bool _isEnabled;
+
+    internal DefaultSentryStructuredLogger(IHub hub, IInternalScopeManager scopeManager, SentryOptions options, ISystemClock clock)
+    {
+        _hub = hub;
+        _scopeManager = scopeManager;
+        _options = options;
+        _clock = clock;
+
+        _isEnabled = options is { Experimental.EnableLogs: true };
+    }
+
+    private protected override void CaptureLog(SentryLogLevel level, string template, object[]? parameters, Action<SentryLog>? configureLog)
+    {
+        if (!_isEnabled)
+        {
+            return;
+        }
+
+        var timestamp = _clock.GetUtcNow();
+
+        if (!TryGetTraceId(_hub, _scopeManager, out var traceId))
+        {
+            _options.DiagnosticLogger?.LogWarning("TraceId not found");
+        }
+
+        _ = TryGetParentSpanId(_hub, _scopeManager, out var parentSpanId);
```

Comment:
Trace-id above has a parent-id. I'm confused why this is split in 2 operations. We want one trace id but possibly different parents?